ui/app/helpers/mountable-secret-engines.js

```diff
@@ -158,6 +164,8 @@ export function allEngines() {
   return [...MOUNTABLE_SECRET_ENGINES, ...ENTERPRISE_SECRET_ENGINES];
 }
 
+export const allEnginesArray = allEngines();
```

Comment:
Thoughts and feelings. We invoke the method `allEngines()` very often in the UI. However, I did not want to invoke this within a for loop (see [here](https://github.com/hashicorp/vault/pull/28951/files#diff-fb429dc7cf4a5d944e2fdb1180845268804caba94579e0db5ab52c3481e3f7c4R42).

I suspect there is some cleanup to be done in which we unintentionally invoke this function within loops, thus invoking it multiple times over when we just need it once. I'll create a a cleanup tasks if folks thinks this is a good pattern to pursue.